src/executor/execute.rs

```diff
@@ -60,9 +66,21 @@ pub fn execute<T: 'static + Debug, U: Store<T> + StoreMut<T> + AlterTable>(
     };
 
     match prepared {
-        Prepared::Create(schema) => storage
-            .insert_schema(&schema)
-            .map(|(storage, _)| (storage, Payload::Create)),
+        Prepared::Create {
+            schema,
+            if_not_exists,
+        } => {
+            if storage.fetch_schema(&schema.table_name).is_ok() {
+                return if if_not_exists {
+                    Ok((storage, Payload::Create))
+                } else {
+                    Err((storage, ExecuteError::TableAlreadyExists.into()))
```

Comment:
👍 I like this.
You transformed `StoreError::SchemaNotFound` to a different and more informative error.
I haven't thought about this before but it looks necessary to do, thanks :)